Source/MaestroBurner.cpp

```diff
@@ -0,0 +1,386 @@
+                     const Real time_in, const Vector<MultiFab>& source) {
+    // timer for profiling
+    BL_PROFILE_VAR("Maestro::BurnerSDC()", BurnerSDC);
+
+    // Put tempbar_init on cart
+    Vector<MultiFab> p0_cart(finest_level + 1);
+    const auto ispec_threshold = network_spec_index(burner_threshold_species);
+
+    if (spherical) {
+        for (int lev = 0; lev <= finest_level; ++lev) {
+            p0_cart[lev].define(grids[lev], dmap[lev], 1, 0);
+            p0_cart[lev].setVal(0.);
+        }
+        Put1dArrayOnCart(p0, p0_cart, false, false, bcs_f, 0);
+    }
+
+    for (int lev = 0; lev <= finest_level; ++lev) {
+        // create mask assuming refinement ratio = 2
+        int finelev = lev + 1;
+        if (lev == finest_level) finelev = finest_level;
+
+        const BoxArray& fba = s_in[finelev].boxArray();
+        const iMultiFab& mask = makeFineMask(s_in[lev], fba, IntVect(2));
+
+        // loop over boxes (make sure mfi takes a cell-centered multifab as an argument)
+#ifdef _OPENMP
+#pragma omp parallel
+#endif
+        for (MFIter mfi(s_in[lev], TilingIfNotGPU()); mfi.isValid(); ++mfi) {
+            // Get the index space of the valid region
+            const Box& tileBox = mfi.tilebox();
+
+            const bool use_mask = !(lev == finest_level);
+
+            const Array4<const Real> s_in_arr = s_in[lev].array(mfi);
+            const Array4<Real> s_out_arr = s_out[lev].array(mfi);
+            const Array4<const Real> p0_cart_arr = p0_cart[lev].array(mfi);
+            const Array4<const Real> source_arr = source[lev].array(mfi);
+            const Array4<const int> mask_arr = mask.array(mfi);
+
+            const auto p0_arr = p0.const_array();
+
+            ParallelFor(tileBox, [=] AMREX_GPU_DEVICE(int i, int j, int k) {
+                if (use_mask && mask_arr(i, j, k))
+                    return;  // cell is covered by finer cells
+
+                auto r = (AMREX_SPACEDIM == 2) ? j : k;
+
+                Real sdc_rhoX[NumSpec];
+                for (int n = 0; n < NumSpec; ++n) {
+                    sdc_rhoX[n] = source_arr(i, j, k, FirstSpec + n);
+                }
+                auto sdc_rhoh = source_arr(i, j, k, RhoH);
+                auto sdc_p0 = spherical ? p0_cart_arr(i, j, k) : p0_arr(lev, r);
+
+                auto rho_in = s_in_arr(i, j, k, Rho);
+                Real rhoX_in[NumSpec];
+                for (int n = 0; n < NumSpec; ++n) {
+                    rhoX_in[n] = s_in_arr(i, j, k, FirstSpec + n);
+                }
+#if NAUX_NET > 0
+                Real aux_in[NumAux];
+                for (int n = 0; n < NumAux; ++n) {
+                    aux_in[n] = s_in_arr(i, j, k, FirstAux + n);
+                }
+#endif
+                auto rhoh_in = s_in_arr(i, j, k, RhoH);
+
+                Real x_test = (ispec_threshold > 0)
+                                  ? rhoX_in[ispec_threshold] / rho_in
+                                  : 0.0;
+
+                sdc_t state_in;
```

Comment:
I don't think we have an sdc_t anymore

Reply:
what is used now?

Reply:
it is just `burn_t`.  It has all of the fields that `sdc_t` had.